File: 프로그래머스/unrated/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    
    c1_idx, c2_idx = 0, 0
    for word in goal:
        if c1_idx < len(cards1) and cards1[c1_idx] == word:
            c1_idx += 1
        elif c2_idx < len(cards2) and cards2[c2_idx] == word:
            c2_idx += 1
        else:
            return "No"
    
    return "Yes"

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        String answer = "";
        int c1 = 0, c2 = 0;
        
        for (String g: goal){
            if (c1 < cards1.length && cards1[c1].equals(g)) {
                c1++;
            }
            else if (c2 < cards2.length && cards2[c2].equals(g)) {
                c2++;
            }
            else return "No";
        }
        
        return "Yes";
    }
}